splink/input_column.py

```diff
@@ -1,112 +1,153 @@
     """
-    return " ".join(n[0].key for n in tree.walk())
 
+    name: str
+    table: str = None
+    quoted: bool = True
+    column_reference: exp.Identifier = None  # the JSON index in 'surname['lat']'
+
+    def unquote(self):
+        return replace(self, quoted=False)
+
+    def exclude_expressions(self):
+        return replace(self, return_column_with_expressions=False)
+
+    def change_name(self, new_name: str):
+        return replace(self, name=new_name)
 
-def add_suffix(tree, suffix) -> Expression:
-    tree = tree.copy()
-    identifier_string = tree.find(exp.Identifier).this
-    identifier_string = f"{identifier_string}{suffix}"
-    tree.find(exp.Identifier).args["this"] = identifier_string
-    return tree
+    def add_prefix(self, prefix: str):
+        return replace(self, name=prefix + self.name) if prefix else self
 
+    def add_suffix(self, suffix: str):
+        return replace(self, name=self.name + suffix) if suffix else self
 
-def add_prefix(tree, prefix) -> Expression:
-    tree = tree.copy()
-    identifier_string = tree.find(exp.Identifier).this
-    identifier_string = f"{prefix}{identifier_string}"
-    tree.find(exp.Identifier).args["this"] = identifier_string
-    return tree
+    def add_table(self, table: str):
+        return replace(self, table=table) if table else self
 
+    def add_alias(self, alias: str) -> exp.Alias:
+        """Alias expects a SQLglot Identifier class or a str.
 
-def add_table(tree, tablename) -> Expression:
-    tree = tree.copy()
-    table_identifier = exp.Identifier(this=tablename, quoted=True)
-    identifier = tree.find(exp.Column)
-    identifier.args["table"] = table_identifier
-    return tree
+        For safety, only pass strings to ensure we capture the entire
+        column reference, not just the identifier.
 
+        For example, col['lat'] is made up of both an identifier and bracket
+        index; the latter of which would be lost if passing identifiers.
+        """
+        # The identifier should be quoted or unquoted before being passed.
+        # This is to ensure we get the correct output.
+        alias = exp.Identifier(this=alias, quoted=self.quoted)
+        return self.build_column_tree().as_(alias)
 
-def remove_quotes_from_identifiers(tree) -> Expression:
-    tree = tree.copy()
-    for identifier in tree.find_all(exp.Identifier):
-        identifier.args["quoted"] = False
-    return tree
+    def add_column_reference(self, index_name: str | int):
+        """
+        Manually add a column reference to your column - ['idx'], for example.
+
+        This is useful during the parsing process or when you try to combine
+        illegal name and bracket combinations such as 'SUR NAME['lat']',
+        without needing to add quotes to the column identifier.
+        """
+        is_str = type(index_name) == str
+        self.column_reference = exp.Literal(this=index_name, is_string=is_str)
+        return self
+
+    def build_column_tree(self):
```

Comment:
I think this should be called `build_sqlglot_expression_tree` or `as_sqlglot_expression_tree`